packages/trace-viewer/src/ui/attachmentsTab.tsx

```diff
@@ -20,9 +20,52 @@ import { ImageDiffView } from '@web/shared/imageDiffView';
 import type { MultiTraceModel } from './modelUtil';
 import { PlaceholderPanel } from './placeholderPanel';
 import type { AfterActionTraceEventAttachment } from '@trace/trace';
+import { CodeMirrorWrapper } from '@web/components/codeMirrorWrapper';
+import { isTextualMimeType } from '@isomorphic/mimeType';
+import { Expandable } from '@web/components/expandable';
+
 
 type Attachment = AfterActionTraceEventAttachment & { traceUrl: string };
 
+type TextAttachmentProps = {
+  attachment: Attachment;
+  label?: string;
+};
+
+const TextAttachment: React.FunctionComponent<TextAttachmentProps> = ({ attachment, label }) => {
+  const [text, setText] = React.useState<string | null>(null);
+
+  React.useEffect(() => {
+    let isMounted = true;
```

Comment:
Instead of this, I wonder if we should change the `<Expandable>` contract to be

`<Expandable expanded={} body={() => <TextAttachment/>}>`

We don't seem to have any usages of it yet, so easy to change.

Reply:
> Instead of this, I wonder if we should change the <Expandable> contract to be
>
> `<Expandable expanded={} body={() => <TextAttachment/>}>`

Not a React expert but I'm not sure I really like this change in contract vs. directly "projecting" the body into the `Expandable` component.

> We don't seem to have any usages of it yet, so easy to change.

The `Expandable` was already used [here](https://github.com/alvaromartmart/playwright/blob/700d06170d2a5d84a989bc5c0738fc539889932d/packages/trace-viewer/src/ui/uiModeFiltersView.tsx#L44-L55), though it doesn't rely on the body, it seems it's only used to toggle the state.

All things considered, I can apply the change if you want as it seems quite straightforward one. Just not sure it's worth it. WDYT?

Reply:
It would look better than the manual isMounted management to me!

Reply:
Got rid of the `isMounted` as part of https://github.com/microsoft/playwright/pull/31215#issuecomment-2198347800, but I'm not sure if your request here is still relevant.

On a different note, I wonder if at this point I should merge `ExpandableAttachment` and `TextAttachment` into a single component for simplicity. Please LMKWYT